routing/router.go

```diff
@@ -2249,3 +2249,288 @@ func generateBandwidthHints(sourceNode *channeldb.LightningNode,
 
 	return bandwidthHints, nil
 }
+
+// runningAmounts keeps running amounts while the route is traversed.
+type runningAmounts struct {
+	// amt is the intended amount to send via the route.
+	amt lnwire.MilliSatoshi
+
+	// max is the running maximum that the route can carry.
+	max lnwire.MilliSatoshi
+}
+
+// checkOrUpdateAmt checks or updates the running amounts while
+// traversing the channels.
+func (r *ChannelRouter) checkOrUpdateAmt(policy *channeldb.ChannelEdgePolicy,
```

Comment:
Nit: doesn't need to be a method of `ChannelRouter`.

Reply:
Made it a method of runningAmts.